File: jpa-example/src/main/java/example/jpa/model/SerialId.java
package example.jpa.model;

import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.MappedSuperclass;
import lombok.Getter;

@Getter
@MappedSuperclass
public class SerialId {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
}

File: jimmer-example/src/main/kotlin/example/jimmer/model/SerialId.kt
package example.jimmer.model

import org.babyfish.jimmer.sql.GeneratedValue
import org.babyfish.jimmer.sql.GenerationType.IDENTITY
import org.babyfish.jimmer.sql.Id
import org.babyfish.jimmer.sql.MappedSuperclass

@MappedSuperclass
interface SerialId {
    @Id
    @GeneratedValue(strategy = IDENTITY)
    val id: Long
}